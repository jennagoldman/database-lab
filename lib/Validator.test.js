const { Validator } = require('./Validator');

describe('Validator class', () => {
  it('can validate data', () => {
    const nameValidatorRequired = new Validator('name', {
      type: String,
      required: true
    });

    const nameValidatorNotRequired = new Validator('name', {
      type: String,
      required: false
    });

    const cat = {
      name: 'Loki',
      age: 4,
      weight: '11 lbs'
    };

    const noNameCat = {
      age: 10,
      weight: '12 lbs'
    };

    expect(nameValidatorRequired.validate(cat)).toEqual('Loki');
    expect(nameValidatorNotRequired.validate(noNameCat)).toEqual(null);
    expect(() => nameValidatorRequired.validate(noNameCat)).toThrowErrorMatchingSnapshot();
  });
});
