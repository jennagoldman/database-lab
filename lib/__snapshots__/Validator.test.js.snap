// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validator class can validate data 1`] = `undefined`;
